_posts/2019/2019-12-01-ensure-you-correctly-build-caching-keys.md

```diff
@@ -0,0 +1,110 @@
+---
+title: "Ensure you correctly build your caching keys"
+description: "Why won't this model update? Oh."
+layout: article
+category: ruby
+image:
+  base: '2019/ensure-you-correctly-build-caching-keys'
+  alt: 'Keys'
+  credit:  Chunlea Ju
+  source: "https://unsplash.com/photos/8fs1X0JFgFE"
+
+---
+
+Caching is a hugely powerful tool in maintaining performance of often requested pages and partials in your application.
+
+It can also be a cause of confusing behaviours (mostly “stuff not updating on the page”) when you test your app in a production environment.
+
+Rails has great built-in support for view fragment caching. The framework includes an elegant way of invoking and busting the cache based on the model `id`, it’s `updated_at` timestamp and an automatically generated digest of the specific view template. You can find more details in the [Rails guide on caching](https://guides.rubyonrails.org/caching_with_rails.html#fragment-caching).
```

Comment:
Mmm — maybe explain what this use or just literally explain "view fragment caching" instead of using the phrase.

"invoking and busting the cache" — I do understand, and maybe most will, but feels like a lot of jargon, esp. next to "view fragment caching".

*its `updated_at`

- Comma before "and"... 😌